src/lib/escape-highlight.ts

```diff
@@ -13,7 +14,7 @@ export const TAG_REPLACEMENT = {
   highlightPostTag: '</mark>',
 };
 
-function replaceTagsAndEscape(value) {
+function replaceTagsAndEscape(value: string): string {
```

Comment:
I'm still not convinced of writing the return type here, it can easily be inferred from the fact that .replace is called on escape (which always returns a string)

Reply:
We explained it here: https://github.com/algolia/instantsearch.js/pull/3617#discussion_r269201029.

Let's chat about it IRL if you're not convinced or need more details.

Reply:
One of the purposes of the types is to reduce the cognitive overhead that reading code might produce. It's also about the contract, with the type definition the intent of the function is clearer.